ledger/model/contract_execution_result.go

```diff
@@ -0,0 +1,62 @@
+package model
+
+import (
+	"github.com/scalar-labs/dl"
+	"github.com/scalar-labs/dl/ledger/asset"
+)
+
+// ContractExecutionResult defines the result of a contract execution.
+// It contains the result of the contract execution along with a list of asset proofs from Ledger and Auditor.
+type ContractExecutionResult struct {
+	Result        dl.JSONObject
+	Proofs        []asset.Proof
+	AuditorProofs []asset.Proof
+}
+
+// Equal checks if two contract execution results have the same values.
+func (r ContractExecutionResult) Equal(another ContractExecutionResult) (equal bool) {
+	var (
+		myResult      dl.JSONObject = r.Result
+		anotherResult dl.JSONObject = another.Result
+	)
+
+	if (myResult == nil && anotherResult != nil) || (myResult != nil && anotherResult == nil) {
+		return false
+	}
+
+	if !myResult.Equal(anotherResult) {
+		return false
+	}
+
+	var (
+		myProofs      []asset.Proof = r.Proofs
+		anotherProofs []asset.Proof = another.Proofs
+	)
+
+	if len(myProofs) != len(anotherProofs) {
+		return false
+	}
+
+	for i := range myProofs {
+		if !myProofs[i].Equal(anotherProofs[i]) {
+			return false
+		}
+	}
+
+	var (
+		myAuditorProofs      []asset.Proof = r.AuditorProofs
+		anotherAuditorProofs []asset.Proof = another.AuditorProofs
+	)
+
+	if len(myAuditorProofs) != len(anotherAuditorProofs) {
+		return false
+	}
+
+	for i := range myAuditorProofs {
+		if !myAuditorProofs[i].Equal(anotherAuditorProofs[i]) {
```

Comment:
I think they might not be sorted in the same order, so I think we should make it ordering-independent?
(The Java version creates a map for one side, and looks up the map to match)

Reply:
@feeblefakie 
Oh, thank you for your reminder.
I didn't realize this.

I use the map to compare too in https://github.com/scalar-labs/scalardl-go-client-sdk/pull/14/commits/02660a70b4741f53bf0572c3dc24f2cfed8fe881

Please check it.
Thank you :)